libraries/AP_HAL_Linux/VideoIn.cpp

```diff
@@ -235,6 +255,48 @@ void VideoIn::prepare_capture()
     }
 }
 
+void VideoIn::crop_image_grey(uint8_t *buffer, uint32_t buffer_size, uint8_t *new_buffer, uint32_t width,
+    uint32_t height, uint32_t cropLeft, uint32_t cropRight, uint32_t cropTop, uint32_t cropBottom)
+{
+    uint32_t i, new_buffer_position = 0, columnCount = 0, lineCount = 0;
+
+    for (i = 0; i < buffer_size; i++) {
+        if (columnCount == width) {
+            columnCount = 0;
+            lineCount++;
+        }
+
+        if (columnCount < cropLeft || columnCount >= (width - cropRight)) {
+            columnCount++;
+            continue;
+        }
+
+        if (lineCount < cropTop || lineCount >= (height - cropBottom)) {
+            columnCount++;
+            continue;
+        }
+
+        columnCount++;
+
+        new_buffer[new_buffer_position] = buffer[i];
+
+        new_buffer_position++;
+    }
+}
```

Comment:
We can use this v4l2 notation, actually I tried this way but the crop didn't work either.